### PR TITLE
Protect against aggregating problematically large documents (part 1)

### PR DESCRIPTION
**Pull request by SethSmucker**
Part of #3656

Document aggregation in KeyToDocumentData.collectDocumentAttributes is unbounded. An exceptionally large document gets buffered in its entirety and can exhaust tserver memory, and the existing doc.agg.threshold stopwatch only logs a warning after the fact. I added a doc.agg.max.entries option (default -1, disabled) and an EventDataQueryEntryLimitFilter that wraps whatever event evaluation filter the query already uses. Once a document hits the limit, the filter rejects the remaining entries, adds a synthetic INCOMPLETE_DOCUMENT marker field so the truncated document is identifiable in results, and seeks past the rest of the document so the excess entries are not read at all.

This is the first tier of the issue and covers standard event queries only. The TLD, ancestor, and parent iterators supply their own evaluation filters and are left for later tiers, along with the proposed non-query-field reductions. New tests cover the filter behavior and the config round-trip.

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java [hunk 1]
```diff
@@ -2551,6 +2551,7 @@ protected void configureSeekingOptions(IteratorSetting cfg, ShardQueryConfigurat
 
         if (config.isSeekingEventAggregation()) {
             addOption(cfg, QueryOptions.SEEKING_EVENT_AGGREGATION, String.valueOf(config.isSeekingEventAggregation()), false);
+            addOption(cfg, QueryOptions.DOC_AGGREGATION_MAX_ENTRIES, String.valueOf(config.getDocAggregationMaxEntries()), false);
```
Comment (apmoriarty) [inline warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java]: This option should be independent of a seeking aggregation.

**Comment by apmoriarty** [comment]
I see this is marked as part one. Please include an integration test that uses AbstractQueryTest and AbstractIngest to demonstrate the event short circuit behavior. 

It's fine if the same large event is ingested repeatedly with different event ids. 

Tests should demonstrate queries that hit entirely within the aggregation window, and also include queries with terms that fall outside the aggregation window (and then document results not coming back, or document an exception to the max size threshold).

File: warehouse/query-core/src/test/java/datawave/query/LargeDocumentProtectionIT.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+     */
+    @Test
+    public void testIndexedTermOutsideWindowDropsDocument() throws Exception {
+        maxEntries = LIMIT;
+        givenDate("20260708");
+        givenQuery("ZEBRA == 'stripes'");
+        expectPlan("ZEBRA == 'stripes'");
+        expectResultCount(0);
+        planAndExecuteQuery();
+    }
```
Comment (apmoriarty) [inline warehouse/query-core/src/test/java/datawave/query/LargeDocumentProtectionIT.java:224]: This is perfect, exactly what the edge case I wanted to see demonstrated in a test
